pipeline/backend/dummy/dummy.go

```diff
@@ -0,0 +1,230 @@
+// Copyright 2024 Woodpecker Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build integration || test
+// +build integration test
+
+package dummy
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/rs/zerolog/log"
+	"github.com/urfave/cli/v2"
+
+	backend "go.woodpecker-ci.org/woodpecker/v2/pipeline/backend/types"
+)
+
+type dummy struct {
+	kv sync.Map
+}
+
+const (
+	// Step names to control step behavior of dummy backend.
+	StepStartFail   = "step_start_fail"
+	StepExecError   = "step_exec_error"
+	EnvKeyStepSleep = "SLEEP"
+	EnvKeyStepType  = "EXPECT_TYPE"
+
+	// Internal const.
+	stepStateStarted   = "started"
+	stepStateDone      = "done"
+	testServiceTimeout = 1 * time.Second
+)
+
+// New returns a dummy backend.
+func New() backend.Backend {
+	return &dummy{
+		kv: sync.Map{},
+	}
+}
+
+func (e *dummy) Name() string {
+	return "dummy"
+}
+
+func (e *dummy) IsAvailable(_ context.Context) bool {
+	return true
```

Comment:
Maybe use:

```suggestion
	return testing.Testing() || os.Getenv("WOODPECKER_ENABLE_DUMMY_BACKEND") == "true"
```

Reply:
well we already put it behind an build tag ... at this point it's overprotective and hinders developers more than as it should ...